src/ucp/api/ucp_def.h

```diff
@@ -190,6 +190,11 @@ typedef struct ucp_mem_attr {
      * Size of the memory segment.
      */
      size_t                 length;
+
+    /**
+     * Virtual address of device memory buffer.
+     */
+     uint64_t               remote_addr;
```

Comment:
should not be exposed to user all together. Please see verbs API.

Reply:
0 offset from end of the struct ? Plus they have configuration for alignment. We would have to brainstorm alignments constrains. Different specs (GPUs, CPUs, IBTA, OPA) have different requirements. BTW, "verbs" level of APIs should rather map to UCT.

Reply:
when doing remote access, need to provide remote address. In verbs, it would be the offset into the allocated memory, according to the remote key.